pyproject.toml

```diff
@@ -34,6 +34,7 @@ dependencies = [
     "toml>=0.10.0,<1",
     "tqdm>=4,<5",
     "typeguard>=4.0.0",
+    "mcp>=1.23.0",
```

Comment:
We should not add this to main dependencies list, instead it should be in the `mcp` group, where fastapi_mcp is being installed. 

The `fastapi_mcp` is the one that is installing mcp dependency 1.12.4 version. So mcp group now should look like:
```
mcp = ["mcp>=1.23.0", "fastapi_mcp"]
```